native/src/main/java/io/ballerina/lib/data/xmldata/FromString.java

```diff
@@ -104,6 +108,22 @@ public static Object fromStringWithType(BString string, Type expType) {
         }
     }
 
+    private static Object stringToFiniteType(String value, FiniteType finiteType) {
+        return finiteType.getValueSpace().stream()
+                .filter(finiteValue -> !(convertToSingletonValue(value, finiteValue) instanceof BError))
```

Comment:
Instead of using `instanceof`, can we use tags or another method?

Reply:
As I know Instanceof is the most preferable/best way now